.github/workflows/main.yml

```diff
@@ -18,25 +18,30 @@ jobs:
         with:
           python-version: '3.10'
 
-      - name: Install dependencies
-        run: |
-          pip install wheel
-          pip install -r requirements.txt
-          pip install -e .
-
-      - name: Check formatting with Black
-        run: |
-          black --check --exclude "(src|alembic)/" .
-
-      - name: Run headless tests
-        uses: coactions/setup-xvfb@v1
+      - uses: snok/install-poetry@v1
         with:
-          run: python -m pytest
-          working-directory: ./  # Optional: Specify the working directory if needed
-          options:  # Optional: Add any additional options or arguments for pytest
+          version: 1.5.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
 
-      - name: flake8 Lint
-        uses: py-actions/flake8@v2
+      - name: Cache deps
+        id: cache-deps
+        uses: actions/cache@v2
         with:
-          plugins: "flake8-docstrings"
-          extra-args: "--docstring-convention=google --exclude=alembic/versions"
+          path: .venv
```

Comment:
Is this necessary with `poetry`?

Reply:
This is necessary for caching the dependencies using the `venv` environment generated from `snok/install-poetry@v1`